// ng-social-links/esm2015/lib/ng-social-links.service.js
import { Injectable, Inject } from '@angular/core';
import { DOCUMENT } from '@angular/common';
import { DEFAULT_SHARE_URL_CONFIG_TOKEN } from './ng-social-links.tokens';
import * as utils from './ng-social-links.utils';
import * as i0 from "@angular/core";
import * as i1 from "@angular/common";
import * as i2 from "./ng-social-links.tokens";
export class NgSocialLinksService {
    constructor(document, defaultConfig) {
        this.document = document;
        this.defaultConfig = defaultConfig;
    }
    getShareLink(provider, config) {
        const url = (config === null || config === void 0 ? void 0 : config.url) || this.defaultConfig.url || this.document.location.origin;
        const title = (config === null || config === void 0 ? void 0 : config.title) || this.defaultConfig.title;
        const description = (config === null || config === void 0 ? void 0 : config.description) || this.defaultConfig.description;
        return utils.getShareLink(provider, { url, title, description });
    }
}
NgSocialLinksService.ɵprov = i0.ɵɵdefineInjectable({ factory: function NgSocialLinksService_Factory() { return new NgSocialLinksService(i0.ɵɵinject(i1.DOCUMENT), i0.ɵɵinject(i2.DEFAULT_SHARE_URL_CONFIG_TOKEN)); }, token: NgSocialLinksService, providedIn: "root" });
NgSocialLinksService.decorators = [
    { type: Injectable, args: [{
                providedIn: 'root'
            },] }
];
NgSocialLinksService.ctorParameters = () => [
    { type: Document, decorators: [{ type: Inject, args: [DOCUMENT,] }] },
    { type: undefined, decorators: [{ type: Inject, args: [DEFAULT_SHARE_URL_CONFIG_TOKEN,] }] }
];
